### DISCUSSION TITLE
my-site/admin is blank

### DISCUSSION THREAD

**Discussion by palomachiara** [tinacms/tinacms#3914] [Help] [answered]
Hello everyone.
I need help with my Forestry to Tina cms migration. 
TinaCMS seems to be set up,
but when I go to my-site/admin/ (https://kaliriu.com /admin/), it shows blank,
I’m thinking it could be an issue with the build command,
I’m using “tinacms build && hugo” (as the build command in Netlify and also in the package.json file). Any thoughts ? thank you in advance. 

My code:
GitHub - palomachiara/kaliriu

**Comment by storynory** [comment]
I had this problem too.  I solved it by deleting the .gitignore file inside the admin folder.  But whether or not you should keep it depends on how you are deploying the side.   Also you have to be sure that your build can pick up your environment tokens.  I have found it is all quite fiddly.  Hope this helps (i'm no expert, just trying out things myself too).

**Reply by palomachiara** [reply]
What do you mean by "you have to be sure that your build can pick up your environment tokens"? How do I do that? I tried deleting the .gitignore file but that didn't work

**Comment by storynory** [comment]
Yesterday I had the same problem and it was tricky, I have to admit.  I am describing what I did and hoping that it is useful. 

When I ran build locally, and then uploaded the whole site via github, it was important that I did not have the .gitignore in the admin folder so that the admin folder and its contents went up too.  If you don't do that, you get a blank page in admin .But if you are building tinacms in the cloud or on the server, then you don't need to delete .gitignore. 

Another mistake I made was that I renamed the environment variables.  The names of the variables in .env have to match the process variables in the tina config file.  I hope that makes sense.   it was a silly mistake I made.  It's best to leave them the same as in the examples / starters. 

 You should be sure to add .env to .gitignore so it does not go into github where people can see your tokens.   If you are building tina on the server, you need to upload the .env file separately, put them in the root of the project, and make sure it can't be accessed via the web.   I was learning about this yesterday too.  Perhaps a bit naive of me. 
 
 Just again, I'm just trying to be helpful based on my experience.  There are people better qualified than me.   I do think it's helpful to let the TInacms developers know that we find this all quite tricky and it isn't all plain sailing. Maybe the documentation could be a little clearer on connecting to the Tina cloud. 

I hope this helps, but in general you got to make sure that the admin folder gets the right contents one way or another.  It needs to have an index file  with links to js files and an assets folder.

**Reply by palomachiara** [reply]
Thank you for your help. But it hasn’t worked for me

**Reply by jamespohalloran** [reply]
When I go to your site's admin URL, I'm prompted to login to TinaCMS, does this mean it's not blank for you anymore?
https://kaliriu.com/admin/

<img width="1117" alt="Screen Shot 2023-05-08 at 9 12 45 AM" src="https://user-images.githubusercontent.com/3323181/236820821-8c13dc9d-7bba-4d27-9cbe-f0905ff623f0.png">

Side note, we generally don't recommend removing the .gitignore from the tina directory. This is a workaround for not building the admin in CI, but it may lead to issues if you make a chance to your schema, and forget to build/push your admin

**Reply by jamespohalloran** [reply]
Actually, I see this admin is deploying with "local mode". That tells me that you're not deploying the production admin in CI. (Since your removed your .gitignore, and you're likely running tinacms dev locally, you're pushing up your admin configured to your localhost URLs)

**Reply by palomachiara** [reply]
It still shows blank to me.. I removed only the .gitignore file inside the tina folder, so that _generated_ would push to my github. I've carefully followed all of the steps from the Migrate from forestry to TinaCMS article, and it still isn't working.. something’s missing here

**Reply by jamespohalloran** [reply]
https://www.loom.com/share/db5bfdb51c534791b5e101b73b55ca33

This make sense? I know you were originally seeing issues with the .gitignore there, but we can try troubleshooting that once we have that proper admin/index.html file on your site.

**Reply by palomachiara** [reply]
Thank you for the video explanation, I appreciate it a lot.
I never deleted that .gitignore file though, but I played around with it anyway- I made sure it didn't push the index.html file, and now it isn't on my git all. But my website is now showing a page not found error, when going to /admin. It doesn't look like it's rebuilding the correct index.html file on its own

**Reply by jamespohalloran** [reply]
Gotcha! Well that's one step closer. This tells us that for some reason, tinacms build either isn't getting run in CI or it's building to the wrong directory. 

A few potential issues:

I see in your package.json your missing `hugo` from your build script: https://github.com/palomachiara/kaliriu/blob/main/package.json#L8

You mentioned this is your build in Netlify:
`tinacms build && hugo`

tinacms likely isn't installed on the build image, so this would either need to be `npm run build` or `npx tinacms build && hugo`

**Reply by palomachiara** [reply]
I've tried both, npm run build and npx tinacms build && hugo, also erased cache, still isn't working :(

**Reply by jamespohalloran** [reply, marked as answer]
Aha, found the issue: https://github.com/palomachiara/kaliriu/pull/1/files#diff-ab8f79b68b7adff7a07db953bf453f3c5aa6ade98d2b1b67d8432b36392489edR2

You have a build command set in your netlify.toml that is overriding the setting in the Netlify UI

**Reply by palomachiara** [reply]
It works, thank you soo much!! What a relief hha

**Comment by BattleOfPlassey** [comment]
I faced the same, my build on GitHub pages did not show /admin page. This is well documented even on tina.io docs.  https://tina.io/docs/tina-cloud/connecting-site/#1-not-providing-a-custom-build-workflow

if you are building the app on github, write your `.github/workflows/deploy.yml` file as in the comment https://github.com/tinacms/tinasaurus/issues/2#issuecomment-1525029418
